### PR TITLE
iOS: Ensure that task observers for VSync callbacks are not bypassed in VSyncWaiterIOS.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/7433.

Investigation: This is caused by a subtle difference between the way in which iOS and Android manage VSync pulses. On Android, the platform main thread wakes up (using the Choreographer) on each VSync pulse. Then, it posts a task to the UI thread to finally service a callback. On iOS however, the CADisplayLink (Choreographer equivalent) is directly scheduled onto the UI thread. This way, on each VSync pulse, only one thread ever gets woken up. The side effect if that no tasks need to posted and the callback run directly. However, the callback itself may have side effects that are usually resolved using a task observer. This is what was happening in the issue linked. So, no task, no task observer invocation. This patch makes it so that the callback is invoked using PostTask. Note that PostTask is being used only to fire task observer and not actually switching threads. Since we are already on the correct thread, there wont be a syscall to the platform waitset either.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm [hunk 1]
```diff
@@ -52,11 +52,21 @@ - (void)await:(shell::VsyncWaiter::Callback)callback {
...
+  // Note: Even though we know we are on the UI thread already (since the
+  // display link was scheduled on the UI thread in the contructor), we use
+  // the PostTask mechanism because the callback may have side-effects that need
+  // to be addressed via a task observer. Invoking the callback by itself
+  // bypasses such task observers.
+  //
+  // We are not using the PostTask for thread switching, but to make task
+  // observers work.
+  blink::Threads::UI()->PostTask([callback = _pendingCallback]() {
+    callback(ftl::TimePoint::Now());
```
Comment (abarth) [inline shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm]: Why don't we want to use |frame_time| ?

**Comment by abarth** [comment]
LGTM

File: shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm (hunk 1 above)
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm]: I was not sure why we were using it in the first place. So got rid of it.

**Comment by eseidelGoogle** [comment]
I presume we plan to land Yegor's reduction/test from the other bug as well?

**Pull request merged**